### DISCUSSION TITLE
It's not using my GPU at all?

### DISCUSSION THREAD

**Discussion by chesman35** [illsk1lls/ZipRipper#22] [Q&A]
Hello; I have a 5700 XT, Latest drivers. I ran the added AMD-fixer script, got walked through the openCL stuff and on the top it says "[OpenCL AVAILABLE]". Cool. 
Other than the fact it's only using 1 CPU thread, and 0% of my GPU?
OpenCL is definitely avalable, so how do I switch to GPU.

Appologies if it's realy obvious, but I don't realy know much about deep technical things with computers, Im only running this "EZ" script for a basically 1 time use case I need, and it's working, it just seems like it's not using my pc to the fullest.

My ETA is... between the 4th and 10th of july. For refference I started this about 2 hours ago... (0.25% at time of writing) It's the 20th of June... HELP

Edit: The bar at the top is stuck on [initializing] so idk what thats about

**Comment by illsk1lls** [comment]
what file format is it? the top bar is displaying incorrectly if it shows "initializing" while john is running...

**Reply by chesman35** [reply]
![image](https://github.com/illsk1lls/ZipRipper/assets/103326166/e42302b5-fb52-449d-9b45-5ff366d911e4)
![image](https://github.com/illsk1lls/ZipRipper/assets/103326166/22197159-1258-4e36-8f68-572f297a8d38)
The online / offline mode doesnt effect anything, I tried all my faild fixes for my gpu in both modes so it's a non factor

**Comment by illsk1lls** [comment]
Ah, PKZIP doesnt have OpenCL..

The issue you are seeing is a titlebar bug, thank you for the screenshot. I'll check it now, it should be displaying, instead of; Initializing->CPU Mode, and OpenCL Available->OpenCL UNSUPPORTED Filetype

**Reply by chesman35** [reply]
shit... is there anyway to at least split the password list and run another instance so i can multithread it?

**Comment by illsk1lls** [comment]
Script is updated with the statusbar fixes. It should now display correctly for resumed jobs

To use more cores, you may be able to manually use the fork option. But to do so you will need to start over with a new job, settings cannot be modified for a resumed job. 

->Start a job, and while its hashing the file (Creating password hash...) kill it with taskmanager, and navigate to %ProgramFiles%\JtR\Run in CMD prompt, then type the following (changing the zipfile path and name to match the file you are targeting):

`zip2john "C:\Full\Path to File\thisfile.zip">"%ProgramData%\JtR\run\pwhash"`

then

`john --fork=8 --wordlist="%ProgramData%\JtR\run\password.lst" --rules=single,all "%ProgramData%\JtR\run\pwhash"`

lmk

I may add a popup to offer this when in CPU mode in a future update

**Comment by illsk1lls** [comment]
Updated the script, added the ability to split the wordlist for CPU only jobs ;) 

When CPU Mode is detected, the script will offer the user the option to split the wordlist via popup.  (The number of processes is determined by the amount of cores you have)

(There may be up to a 60 second delay from when a password is found by one process, before the remaining processes are halted.. It is also worth noting, prior this change all threads on your processor should have already been in use using OpenMP)

EDIT: I'm still weighing how valuable this is, as wordlists are usually built in order of MostLikely>LeastLikely, not in alphabetical order.. So if a wordlist is built correctly this may actually slow the job down. In a future version I may end up only triggering the popup only when a very large wordlist is used (1gb+).
